rubin_sim/scheduler/features/features.py

```diff
@@ -515,12 +515,17 @@ class NoteLastObserved(BaseSurveyFeature):
         Substring to match an observation note field to keep track of.
     """
 
-    def __init__(self, note):
+    def __init__(self, note, filtername=None):
         self.note = note
+        self.filtername = filtername
         self.feature = None
 
     def add_observation(self, observation, indx=None):
-        if self.note in observation["note"]:
+        if self.note in observation["note"] and (
+            self.filtername == observation["filter"]
+            if self.filtername is not None
+            else True
```

Comment:
That seems like some convoluted-to-read python for a late-at-night at the observatory person. I know it's the same, but is there actually a performance penalty for making it an if/else inside the first "if"? 
ie.. more like 
```
if self.note in observation['note']:  
   if (self.filtername is None) or (self.filtername == observation['filter']):
       self.feature = observation['mjd']
```
(should probably note in the doc string that observation note matches if self.note is contained in the note, not just exact matches .. which 'substring' may be intended to convey, but might not entirely?)

Reply:
good point.. 

I merged your second if into the first, which gives the same outcome:

```
        if self.note in observation["note"] and (
            self.filtername is None or self.filtername == observation["filter"]
        ):

```